### DISCUSSION TITLE
Add to non-existing array

### DISCUSSION THREAD

**Discussion by saadlu** [mikefarah/yq#692] [Q&A] [answered]
Given a yaml

```yaml
spec:
   some: wombat
```

in V3, the following worked

```bash
yq w -i file.yaml 'spec.config[+]' a
```
giving

```yaml
spec:
  some: wombat
  config:
    - a 
```

Subsequently we can add more elements to the array by running the same `yq` command.

What's the corresponding command in V4?

Doing 

```bash
yq e -i '.spec.config + "a"' file.yaml
```
gives

```bash
Error: !!null cannot be added to !!str
```

could do

```bash
yq e -i '.spec.config[0] = a' file.yaml
```

but this requires maintaing the index, no?

**Comment by mikefarah** [comment, marked as answer]
I'd use the alternative operator with add:

```
yq e -i '.spec.config = .spec.config // [] + "a"' file.yaml
```

**Reply by saadlu** [reply]
cool thanks :)

**Comment by saadlu** [comment]
@mikefarah Just discovered that this, although worked in 4.4.0, does not work in 4.5.0, 4.6.0 ..

**Reply by saadlu** [reply]
ah its precedence

to make it work in 4.5.0+ need to do

```bash
yq e -i '.spec.config = (.spec.config // []) + "a"' file.yaml
```

**Comment by mudideng** [comment]
How does the formatting work on this @mikefarah?
If I create an empty array first `yq e -i '.spec.config = []' file.yaml`, it seems like it will always stick to the format 
`[a, b, c]` instead of
```
- a
- b
- c
```
when I later do `yq e -i '.spec.config += "a"' file.yaml`. Is there a way to force the format to the latter?
